### ISSUE TITLE
How to check if the email field is required first

### ISSUE THREAD

**Issue by amine0909** [colinhacks/zod#2408]
Hello guys, 
I'm trying to validate the email field, I want to check if the field is required or not first, then I want to trigger the email validation

```ts
    -------- Validator.ts --------
    export const FormDataValidator = z.object({
      email: z.string({required_error: "Required field"}).email(invalidEmail),
    })

    -------- index.ts ----------
    const result = FormDataValidator.shape.email.safeParse(
        emailReference.value
    );
```

but the problem is that always i get "Invalid Email" from the "email function()" .
Is there any way to fix this?

**Comment by kuriacka** [comment]
Try this:
```
-------- Validator.ts --------
export const FormDataValidator = z.object({
    email: string().min(1, "Required field").email(invalidEmail),
})
 ```

**Comment by amine0909** [comment]
> 

I tried it and i got this result
```
   ZodError: [
  {
    "code": "too_small",
    "minimum": 1,
    "type": "string",
    "inclusive": true,
    "exact": false,
    "message": "Required field",
    "path": []
  },
  {
    "validation": "email",
    "code": "invalid_string",
    "message": "Invalid email",
    "path": []
  }
]
    at get error [as error] (types.js:43:1)
    at account-management.ts:36:30
    at HTMLInputElement.<anonymous> (events.ts:10:60)
```

**Comment by kuriacka** [comment]
I guess that is what you want - its an array of issues, so if you take first element of that array, it should be error you are looking for. However this approach to grabbing issues is quite cumbersome, but without knowing more about implementation its hard to give any better advice.

**Comment by alexrdas** [comment]
![image](https://github.com/colinhacks/zod/assets/98311806/2ac2508e-0d72-424f-96c4-1cfa71eb478a)
para el que lo necesite...

**Comment by byStrange** [comment]
I think zod returns the last validation's error. Like new validation checks overwrites existing validation checks. For example
1.
```javascript
const emailType = z.string().email({ message: "invalid email"}).min(1, "This field is required")
```
2.
```javascript
const emailType z.string().min(1, "This field is required").email({ message: "invalid email"})
```

In the first example you will get desired output. When you try to validate with empty value, you will get `This field is required`, not `invalid email`. But in the second example you will get `invalid email` error. Because both ``min`` and ``email`` validation checks get run and the last one gets applied (I think). That's why the last validation is the one that gets applied at the end overwriting previous errors. 

In summary, the last validation check in Zod takes precedence and its error message will overwrite any previous validation errors.

**Comment by byStrange** [comment]
So option 1 would be a solution for this.
